code/script.js

```diff
@@ -222,104 +223,121 @@ const generateBoard = () => {
   } else {
-    if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
-    } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+    charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+      alert (`No sorry, the person doesn´t have ${value} hair! Remove all people with ${value} hair`)
     }
-  }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
-
-    for accessories and other
+  } else if (category === 'eyes') { 
+     if (keep) {
+      charactersInPlay = charactersInPlay.filter((person) =>  person[category].includes(value))
+       alert (`Oh yes,the person has ${value} eyes! Keep all with ${value} eyes`)
+       
+     } else {
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+       alert (`Nope, sorry, the person has´nt ${value} eyes! Remove all people with ${value}eyes`)   
+    }
+    
+    } else if (category === 'accessories') {
+     if (keep) {
       charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
-      or
+       alert (`Yes the person wears a ${value} Keep all the people that wears a ${value}`)
+
+    } else {
       charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
-  */
+     alert
+     ( `No,the person doesn´t wear ${value}! Remove all people that wear ${value} `)
+     
+   }
+  }   else if (category=== 'other') {
+        if (keep) {
+        charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
+          alert
+          (`Yes the person is a ${value}! Keep all the people that are smokers! ${value}`)
+         } else {
+          charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+           alert
+           (`No the person is not a ${value}! Remove all the people that are smokers!${value}`)
+        }
+      }
 
-  // Invoke a function to redraw the board with the remaining people.
+      generateBoard(keep)
 }
-
-// when clicking guess, the player first have to confirm that they want to make a guess.
+ 
 const guess = (personToConfirm) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
-}
+  let thePlayerGuess = confirm (`Is this your guess ${personToConfirm}?`)
 
-// If you confirm, this function is invoked
+  if (thePlayerGuess) {
+    checkMyGuess(personToConfirm) 
+ 
+  } else {
+    alert (`Sorry, try again!`)
+
+  }  
+} 
```

Comment:
It seems like you use different spaces and indentations in your project. You can use prettier code formatter that helps clean up your code, so you don't have for example unnecessary spaces or improper indentations :)